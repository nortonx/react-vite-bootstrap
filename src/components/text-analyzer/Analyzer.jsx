import Textarea from "../Textarea";
import { useState } from 'react';
import Counter from "./Counter";

const Analyzer = () => {
  const [content, setContent] = useState("");

  function onChangeText(content) {
    setContent(content);
  }

  function wordCounter(content) {
    if (content) {
      const words = content.split(" ");
      return words?.length;
    }
    return 0;
  }

  function characterCounter(content) {
    const characters = content.split("");
    return characters?.length ?? 0;
  }

  function sentencesCounter(content) {
    const sentences = content.match(/[^\.!\?]+[\.!\?]+/g);
    return sentences?.length ?? 0;
  }

  function paragraphsCounter(content) {
    if (content) {
      const paragraphs = content.split(/\r?\n/);
      return paragraphs?.length;
    }
    return 0;
  }

  return(
    <div className="container text-analyzer" data-testid="analyzer">
      <div className="col">
        <div className="row">
          <Counter
            label="Words"
            counter={wordCounter(content)}
          />
          <Counter
            label="Characters"
            counter={characterCounter(content)}
          />
          <Counter
            label="Sentences"
            counter={sentencesCounter(content)}
          />
          <Counter
            label="Paragraphs"
            counter={paragraphsCounter(content)}
          />
        </div>
        <div className="row">
          <Textarea 
            label="Type in your text"
            id="text-analyzer"
            onChangeText={onChangeText}
          />
        </div>
      </div>
    </div>
  );
};

export default Analyzer;