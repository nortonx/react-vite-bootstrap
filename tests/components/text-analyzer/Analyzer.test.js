import { fireEvent, render, screen } from "@testing-library/react";

import Analyzer from "../../../src/components/text-analyzer/Analyzer";

describe("Analyzer", () => {
  beforeEach(() => {
    render(<Analyzer />);
  })
  it("Should render component and match snapshot", async () => {
    const analyzer = await screen.findByTestId("analyzer");
    expect(analyzer).toBeInTheDocument();
    expect(analyzer).toMatchSnapshot();
  });
})