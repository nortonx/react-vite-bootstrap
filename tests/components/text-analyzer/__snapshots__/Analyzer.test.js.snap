// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analyzer Should render component and match snapshot 1`] = `
<div
  class="container text-analyzer"
  data-testid="analyzer"
>
  <div
    class="col"
  >
    <div
      class="row"
    >
      <div
        class="col"
        data-testid="counter"
      >
        <h1
          class="m-2 text-center"
        >
          Words
        </h1>
        <h2
          class="text-center text-body-secondary"
        >
          0
        </h2>
      </div>
      <div
        class="col"
        data-testid="counter"
      >
        <h1
          class="m-2 text-center"
        >
          Characters
        </h1>
        <h2
          class="text-center text-body-secondary"
        >
          0
        </h2>
      </div>
      <div
        class="col"
        data-testid="counter"
      >
        <h1
          class="m-2 text-center"
        >
          Sentences
        </h1>
        <h2
          class="text-center text-body-secondary"
        >
          0
        </h2>
      </div>
      <div
        class="col"
        data-testid="counter"
      >
        <h1
          class="m-2 text-center"
        >
          Paragraphs
        </h1>
        <h2
          class="text-center text-body-secondary"
        >
          0
        </h2>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="textarea mb-3"
      >
        <label
          class="form-label"
          for="text-analyzer"
        >
          Type in your text
        </label>
        <textarea
          class="form-control"
          data-testid="textarea"
          id="text-analyzer"
        />
      </div>
    </div>
  </div>
</div>
`;
